app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtils.kt

```diff
@@ -122,6 +125,63 @@ object FirebaseDatabaseUtils {
         )
     }
 
+    /**
+     * Helper function to obtain the posts of a tournament from the database
+     * @param data the data snapshot containing the posts
+     * @return a list containing the posts of the tournament
+     */
+    fun transformPostList(data: DataSnapshot?): List<TournamentPost> {
+        return data?.children?.mapNotNull {
+            transformPost(it)
+        } ?: emptyList()
+    }
+
+    /**
+     * Helper function to obtain a post from a database snapshot
+     * @param data the data snapshot containing the post
+     * @return the post, or null if an error occurred
+     */
+    fun transformPost(data: DataSnapshot?): TournamentPost? {
+        val userId = data?.child("userId")?.value as String?
+        val run = transformRun(data?.child("run"))
+        val votes = getNumber(data?.child("votes"))?.toInt()
+        val date = transformLocalDateTime(data?.child("date"))
+        // Unchecked cast here but should work without problem
+        val usersVotes =
+            (data?.child("usersVotes")?.value ?: emptyMap<String, Int>()) as Map<String, Int>
+
+        if (userId == null || run == null || votes == null || date == null) {
+            Log.e(this::class.java.name, "TournamentPost had null values")
+            return null
+        }
+
+        return TournamentPost(userId, run, votes, date, usersVotes.toMutableMap())
+    }
+
+    /**
+     * Helper function to obtain an object LocalDateTime from a data snapshot
+     * @param data the data snapshot containing the LocalDateTime
+     * @return the LocalDateTime, or null if an error occurred
+     */
+    fun transformLocalDateTime(data: DataSnapshot?): LocalDateTime? {
+        val year = getNumber(data?.child("year"))?.toInt()
+        val month = getNumber(data?.child("monthValue"))?.toInt()
+        val dayOfMonth = getNumber(data?.child("dayOfMonth"))?.toInt()
+        val hour = getNumber(data?.child("hour"))?.toInt()
+        val minute = getNumber(data?.child("minute"))?.toInt()
+        val second = getNumber(data?.child("second"))?.toInt()
+        val nano = getNumber(data?.child("nano"))?.toInt()
+
+        if (year == null || month == null || dayOfMonth == null || hour == null || minute == null ||
+            second == null || nano == null
+        ) {
```

Comment:
Is there a nicer way to show this condition ?
This is purely about esthetics but that does not look right
```
        if (year == null || month == null || dayOfMonth == null || 
            hour == null || minute == null || second == null || nano == null
        ) {
```

Reply:
I agree, I will change this with your format, thanks.